server/testutil/app/app.go

```diff
@@ -0,0 +1,220 @@
+package app
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	bazelgo "github.com/bazelbuild/rules_go/go/tools/bazel"
+)
+
+const (
+	// readyCheckPollInterval determines how often to poll BuildBuddy server to check
+	// whether it's up and running.
+	readyCheckPollInterval = 500 * time.Millisecond
+	// readyCheckTimeout determines how long to wait until giving up on waiting for
+	// BuildBuddy server to become ready. If this timeout is reached, the test case
+	// running the server will fail with a timeout error.
+	readyCheckTimeout = 30 * time.Second
+)
+
+type App struct {
+	httpPort       int
+	gRPCPort       int
+	monitoringPort int
+	mu             sync.Mutex
+	stdout         bytes.Buffer
+	stderr         bytes.Buffer
+	exited         bool
+	// err is the error returned by `cmd.Wait()`.
+	err error
+}
+
+// appParams are configurable via `RunOpt`s.
+type appParams struct {
+	cmdPath    string
+	configPath string
+	cmdArgs    []string
+}
+
+// Run a local BuildBuddy server for the scope of the given test case.
+//
+// Options may be given to override the default server options.
+func Run(t *testing.T, opts ...RunOpt) *App {
+	p := &appParams{cmdArgs: []string{}}
+	for _, opt := range opts {
+		if err := opt(p); err != nil {
+			t.Fatal(err)
+		}
+	}
+	if p.cmdPath == "" {
+		t.Fatalf("missing `WithBuildBuddyCmdPath` opt")
+	}
+	if p.configPath == "" {
+		t.Fatalf("missing `WithConfigPath` opt")
+	}
```

Comment:
nit: see above -- if these aren't working defaults then we should require them as args to run